### PR TITLE
Allow to set description of request

### PR DESCRIPTION
**Pull request by bzwei**
Addresses the issue that the description field of the request was not saved.
Fixes https://projects.engineering.redhat.com/browse/SSP-454

### CODE REVIEW

**Comment by hsong-rh** [comment]
One more place for the filter needs to be changed:

In `public/approval/v1.0/openapi.json,` all query parameters about `state/decision/requester` on `requests` have to be removed:

https://github.com/ManageIQ/approval-api/blob/master/public/approval/v1.0/openapi.json#L190 to L198;
https://github.com/ManageIQ/approval-api/blob/master/public/approval/v1.0/openapi.json#L987 to L1030

Do you want to add changes here or to open another PR for this?

File: spec/controllers/v1.0/requests_controller_spec.rb [hunk 1]
```diff
@@ -81,7 +81,7 @@
   end
 
   # Test suite for GET /requests?state=
-  describe 'GET /requests?state=notified' do
+  describe 'GET /requests?filter[state]=notified' do
```
Comment (gmcculloug) [inline spec/controllers/v1.0/requests_controller_spec.rb]: In the future these unrelated changes would preferable be moved to a separate PR or at least a separate commit as it is unrelated to the other work in this PR.

File: app/controllers/api/v1x0/requests_controller.rb [hunk 2]
```diff
@@ -28,7 +28,7 @@ def index
       private
 
       def request_params
-        params.permit(:name, :requester, :content => {})
+        params.permit(:name, :description, :requester, :content => {})
```
Comment (gmcculloug) [inline app/controllers/api/v1x0/requests_controller.rb]: @bzwei Are any changes required to `openapi.json`  for this?

Comment (bzwei) [reply app/controllers/api/v1x0/requests_controller.rb]: No. The bug was open because the spec says description is optional, but code ignored it.

**Pull request merged**